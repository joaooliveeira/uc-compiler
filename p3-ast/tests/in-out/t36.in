void f() {
  N++;
  N+=1;
  N%=0;
  N-=100;
  N/=4;
}
