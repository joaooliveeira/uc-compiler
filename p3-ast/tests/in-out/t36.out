Program:
    FuncDef:
        Type: void @ 1:1
        Decl: ID(name=f)
            FuncDecl:
                VarDecl:
                    Type: void @ 1:1
        Compound: @ 1:10
            UnaryOp: p++ @ 2:3
                ID: N @ 2:3
            Assignment: += @ 3:3
                ID: N @ 3:3
                Constant: int, 1 @ 3:6
            Assignment: %= @ 4:3
                ID: N @ 4:3
                Constant: int, 0 @ 4:6
            Assignment: -= @ 5:3
                ID: N @ 5:3
                Constant: int, 100 @ 5:6
            Assignment: /= @ 6:3
                ID: N @ 6:3
                Constant: int, 4 @ 6:6
